contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java

```diff
@@ -301,12 +312,10 @@ protected void replay(long windowId)
   @Override
   public void endWindow()
   {
+    Map<KafkaPartition, Long> carryOn = new HashMap<>(offsetStats);
+    offsetTrackHistory.add(Pair.of(currentWindowId, carryOn));
```

Comment:
This should only be done for the simple consumer as that's the condition to clear the state in committed.